packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -2,20 +2,140 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 import { FtlMsg } from 'fxa-react/lib/utils';
+
 import { ReactComponent as GoogleLogo } from './google-logo.svg';
 import { ReactComponent as AppleLogo } from './apple-logo.svg';
 
+import { useAccount, useConfig, useStorage } from '../../models';
+import Storage from '../../lib/storage';
+
 export type ThirdPartyAuthProps = {
   enabled?: boolean;
 };
 
+type ThirdPartyAuthSigninParams = {
+  authorizationEndpoint: string;
+  clientId: string;
+  scope: string;
+  redirectUri: string;
+  responseType: string;
+  responseMode?: string;
+};
+
+const GOOGLE_SCOPES = 'openid email profile';
+const APPLE_SCOPES = 'email';
+
+/**
+ * ThirdPartyAuth component
+ * A React component that renders Google and Apple third-party authentication buttons.
+ * It handles user sign-in with the respective provider when the buttons are clicked.
+ */
 const ThirdPartyAuth = ({ enabled = false }: ThirdPartyAuthProps) => {
+  const config = useConfig();
+  const storageData = useStorage();
+  const account = useAccount();
+
+  useEffect(() => {
+    // TODO: Figure out why `storageData` is not available
+    const authParams = Storage.factory('localStorage', window).get(
+      'fxa_third_party_params'
+    );
+    if (authParams) {
+      completeSignIn();
+    }
+  }, []);
+
   if (!enabled) {
     return null;
   }
 
+  /**
+   * signIn - Handles the sign-in process for third-party authentication providers.
+   * Creates a form, sets necessary parameters, and submits it to the provider's
+   * authorization endpoint.
+   *
+   * @param {ThirdPartyAuthSigninParams} config - Configuration parameters for the signIn process.
+   */
+  function signIn(config: ThirdPartyAuthSigninParams) {
+    clearStoredParams();
+
+    // We stash originating location in the state oauth param
+    // because we will need it to use it to log the user into FxA
+    const currentParams = new URLSearchParams(window.location.search);
+    currentParams.delete('deeplink');
+    currentParams.set('showReactApp', 'true');
+
+    const state = encodeURIComponent(
+      `${window.location.origin}${
+        window.location.pathname
+      }?${currentParams.toString()}`
+    );
```

Comment:
Not sure if it's any simpler to use `new URL()` directly and tweak the two desired search params:

```js
const url = new URL(window.location);
url.searchParams.delete('deeplink');
url.searchParams.set('showReactApp', 'true');

const state = encodeURIComponent(url.href)
```

Should convert something like this:
``https://accounts.firefox.com/settings/?one=two&deeplink=true``

To this:
`https://accounts.firefox.com/settings/?one=two&showReactApp=true`

Or encoded:
`https%3A%2F%2Faccounts.firefox.com%2Fsettings%2F%3Fone%3Dtwo%26showReactApp%3Dtrue`

I guess the main/only difference is if you want to rebuild the href as a template string in `encodeURIComponent()`. :shrug: